advanced/demos/python_GUI.py

```diff
@@ -0,0 +1,318 @@
+import hebi
+import numpy
+from time import sleep
+import tkinter
+from pyquaternion import Quaternion
+
+
+class Mesh:
+    '''
+    This class describes a 3D mesh that is used to represent the robot components
+    '''
+
+    def __init__(self, vertices, edges, faces, face_vectors):
+        # vertices is a list of points in a mesh
+        self.vertices = vertices
+        # edges is a list of edges that connect adjacent vertices
+        self.edges = edges
+        # faces are a list of faces in the mesh. faces can contain 3 or more points
+        self.faces = faces
+        # face_vectors is a list of unit vectors that are parellel to the corresponding faces. These are used for shading
+        self.face_vectors = face_vectors
+
+
+'''
+these "base" meshes are parameterizable meshes that represent the base shapes before any transformations have been 
+applied. the "@1,@2..." are the parametric variables. These are replaced by the values of your choice by parse_mesh (see
+below). The first is a cube and the second is a rectangle of variable length
+'''
+base_meshes = {'actuator': Mesh(vertices=[['-1*@1', '-1*@1', '@1'], ['-1*@1', '-1*@1', '-1*@1'],
+                                          ['-1*@1', '@1', '@1'], ['-1*@1', '@1', '-1*@1'],
+                                          ['@1', '@1', '-1*@1'], ['@1', '@1', '@1'],
+                                          ['@1', '-1*@1', '@1'], ['@1', '-1*@1', '-1*@1']],
+                                edges=[(0, 1), (0, 6), (0, 2), (1, 7), (1, 3), (2, 3), (2, 5), (3, 4), (4, 5), (4, 7),
+                                       (5, 6), (6, 7)],
+                                faces=[(0, 1, 3, 2), (0, 1, 7, 6), (0, 2, 5, 6), (2, 3, 4, 5), (1, 3, 4, 7),
+                                       (4, 5, 6, 7)],
+                                face_vectors=[(-1, 0, 0), (0, -1, 0), (0, 0, 1), (0, 1, 0), (0, 0, -1), (1, 0, 0)]),
+
+               'link': Mesh(vertices=[[0, '-1*@1', '@1'], [0, '-1*@1', '-1*@1'],
+                                      [0, '@1', '@1'], [0, '@1', '-1*@1'],
+                                      ['@2', '@1', '-1*@1'], ['@2', '@1', '@1'],
+                                      ['@2', '-1*@1', '@1'], ['@2', '-1*@1', '-1*@1']],
+                            edges=[(0, 1), (0, 6), (0, 2), (1, 7), (1, 3), (2, 3), (2, 5), (3, 4), (4, 5), (4, 7),
+                                   (5, 6), (6, 7)],
+                            faces=[(0, 1, 3, 2), (0, 1, 7, 6), (0, 2, 5, 6), (2, 3, 4, 5), (1, 3, 4, 7), (4, 5, 6, 7)],
+                            face_vectors=[(-1, 0, 0), (0, -1, 0), (0, 0, 1), (0, 1, 0), (0, 0, -1), (1, 0, 0)])}
+
+
+# this is used to draw models of robots with a virtual position
+class VirtualFeedback:
+    def __init__(self, position,orientation):
+        self.position = position  # the virtual positions of the robot
+        self.orientation = orientation  # this is the identity quaternion
+
+
+'''
```

Comment:
Small request:

Python is weird with how comments are associated with classes/methods/etc. Generally, the comment corresponding to a class definition, method definition, etc goes immediately below said symbol. So the comment here should appear under the line `class Robot:` like below:

```py
class Robot:
  '''
the Robot class represents your robot. specific values for the robot model are defined in __init__ and can be changed
according to your specific configuration
  '''

 # ....rest of class here....
```